modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -417,76 +417,31 @@ private static string CamelcaseToUnderscore(this string instance, bool lowerCase
         }
 
         /// <summary>
-        /// Performs a case-sensitive comparison to another string, return -1 if less, 0 if equal and +1 if greater.
+        /// Performs a case-sensitive comparison to another string and returns an integer that indicates their relative position in the sort order.
         /// </summary>
         /// <seealso cref="NocasecmpTo(string, string)"/>
         /// <seealso cref="CompareTo(string, string, bool)"/>
         /// <param name="instance">The string to compare.</param>
         /// <param name="to">The other string to compare.</param>
-        /// <returns>-1 if less, 0 if equal and +1 if greater.</returns>
+        /// <returns>An integer that indicates the lexical relationship between the two comparands.</returns>
         public static int CasecmpTo(this string instance, string to)
         {
             return instance.CompareTo(to, caseSensitive: true);
         }
 
         /// <summary>
-        /// Performs a comparison to another string, return -1 if less, 0 if equal and +1 if greater.
+        /// Performs a comparison to another string and returns an integer that indicates their relative position in the sort order.
         /// </summary>
         /// <param name="instance">The string to compare.</param>
         /// <param name="to">The other string to compare.</param>
         /// <param name="caseSensitive">
         /// If <see langword="true"/>, the comparison will be case sensitive.
         /// </param>
-        /// <returns>-1 if less, 0 if equal and +1 if greater.</returns>
+        /// <returns>An integer that indicates the lexical relationship between the two comparands.</returns>
+        [Obsolete("Use string.Compare instead.")]
```

Comment:
It didn't occur to me while reviewing earlier, but this means the calls in `CasecmpTo` and `NocasecmpTo` now warn with `CS0618`. It probably should be silenced.

Reply:
Oh oops, I totally missed this. Yeah, or we can also call the BCL's `string.Compare` directly.